juniper/src/types/async_await.rs

```diff
@@ -258,31 +258,29 @@ where
                 let is_non_null = meta_field.field_type.is_non_null();
 
                 let response_name = response_name.to_string();
-                async_values.push_back(AsyncValueFuture::Field(async move {
-                    // TODO: implement custom future type instead of
-                    //       two-level boxing.
-                    let res = instance
-                        .resolve_field_async(info, f.name.item, &args, &sub_exec)
-                        .await;
-
-                    let value = match res {
-                        Ok(Value::Null) if is_non_null => None,
-                        Ok(v) => Some(v),
-                        Err(e) => {
-                            sub_exec.push_error_at(e, pos);
-
-                            if is_non_null {
-                                None
-                            } else {
-                                Some(Value::null())
-                            }
+                let res = instance
+                    .resolve_field_async(info, f.name.item, &args, &sub_exec)
+                    .await;
```

Comment:
It looks like this `.await` breaks concurrency. The await now happens directly within the `for` loop body, so the future will pause at this point before processing the next select. The intention of `async_values` is that the loop should produce all the futures synchronously up front, then after this loop there is another `while` loop at the bottom of the function which polls the `FuturesOrdered` so that all its contained futures may progress _simultaneously_.